compiler/src/iree/compiler/Dialect/Encoding/IR/EncodingAttrs.td

```diff
@@ -79,19 +79,15 @@ def EncodingAttr :
     - The maps in `user_indexing_maps` should be composable.
   }];
 
-  let assemblyFormat = "`<` struct(params) `>`";
+  let hasCustomAssemblyFormat = 1;
 
   let parameters = (ins
     AttrParameter<"IntegerAttr", "this tensor operand's index in the parameter list">:$operand_index,
     AttrParameter<"EncodingOpTypeAttr", "operand type">:$op_type,
     AttrParameter<"ArrayAttr", "element types of the user's operands">:$element_types,
     OptionalParameter<"ArrayAttr", "Indexing maps of the operation using this tensor">:$user_indexing_maps,
-    // TODO(#19897): Switch round_dims_to to iteration_sizes.
-    // Note that today we abuse the attribute to carry narrow matrix
-    // information. The end goal is deprecating the field and add a
-    // "iteration_sizes" field to describe the shape. It is useful to
-    // handle narrow matrix cases.
-    OptionalParameter<"DenseArrayAttr", "Values for padding M,N,K dimensions">:$round_dims_to,
+    OptionalParameter<"ArrayAttr", "The original operation's iteration sizes. "
+    "This is used for handling narrow matrix cases.">:$iteration_sizes,
```

Comment:
I'd decouple the device requirement and phrase it like `this is useful for handling skinny dimension sizes`.

Reply:
Done